src/Functions/array/arrayDotProduct.cpp

```diff
@@ -385,6 +385,35 @@ using FunctionArrayDotProduct = FunctionArrayScalarProduct<DotProduct>;
 
 REGISTER_FUNCTION(ArrayDotProduct)
 {
+    FunctionDocumentation::Description description = R"(
+Returns the dot product of two arrays.
+
+Aliases: `scalarProduct`, `dotProduct`
+
+:::note
+The sizes of the two vectors must be equal. Arrays and Tuples may also contain mixed element types.
+:::
+)";
+    FunctionDocumentation::Syntax syntax = "arrayDotProduct(v1, v2)";
+    FunctionDocumentation::Arguments arguments = {
+        {"v1", "First vector. [Array](/sql-reference/data-types/array) or [Tuple](../data-types/tuple.md) of numeric values."},
+        {"v2", "Second vector. [Array](/sql-reference/data-types/array) or [Tuple](../data-types/tuple.md) of numeric values."},
+    };
+    FunctionDocumentation::ReturnedValue returned_value = R"(
+The dot product of the two vectors. [Numeric](/native-protocol/columns#numeric-types).
+
+:::note
+The return type is determined by the type of the arguments. If Arrays or Tuples contain mixed element types then the result type is the supertype.
+:::
+)";
+    FunctionDocumentation::Examples examples = {
+        {"Array example", "SELECT arrayDotProduct([1, 2, 3], [4, 5, 6]) AS res, toTypeName(res);", "32    UInt16"},
+        {"Tuple example", "SELECT dotProduct((1::UInt16, 2::UInt8, 3::Float32),(4::Int16, 5::Float32, 6::UInt8)) AS res, toTypeName(res);", "32    Float64"}
+    };
+    FunctionDocumentation::IntroducedIn introduced_in = {1, 1};
```

Comment:
I didn't double check but array functions are quite specialized and dot product as a function is even more specialized. I can imagine that it was not part of v1.1 (which came out 2017 or so). In fact, almost all functions in this PR use 1.1 as version. Can you confirm that this is actually true?

Reply:
This one was a me error, sorry. Should be v23,5